Api/Helper/File.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Werkspot\BingAdsApiBundle\Api\Helper;
+
+use GuzzleHttp\ClientInterface;
+use Symfony\Component\Filesystem\FileSystem;
+use Werkspot\BingAdsApiBundle\Guzzle\Exceptions\CurlException;
+use Werkspot\BingAdsApiBundle\Guzzle\Exceptions\HttpStatusCodeException;
+
+class File
+{
+    /**
+     * @var ClientInterface
+     */
+    private $guzzleClient;
+
+    /**
+     * @var FileSystem
+     */
+    private $fileSystem;
+
+    public function __construct(ClientInterface $guzzleClient = null)
+    {
+        $this->guzzleClient = $guzzleClient;
+        $this->fileSystem = new FileSystem();
+
+    }
+
+    /**
+     * @param $source
+     * @param null|string $destination
+     *
+     * @return bool|string
+     *
+     * @throws \Exception
+     */
+    public function getFile($source, $destination = null)
```

Comment:
This is a weird name, getFile tells me I can get/download a file, but this copies it?

I tried to understand the code, but it took me more then 30 seconds to see what it's doing, this needs clarification, try to avoid the nested if's as well, that would probably help a lot already. But I have the feeling this is doing too much.